### PR TITLE
fix: remove captures from context

### PR DESCRIPTION
**Pull request by Theodus**

### CODE REVIEW

**Comment by Jannis** [comment]
As long as this does not make the performance worse, I'm liking this a lot. It would only slow things down if reusing `Captures` across cost models from different indexers would speed things up. But I would assume that cost models of different indexers are sufficiently different that their captures have little overlap _and_ I'm not even sure if the existing code saves any work if `Captures` is carried over from indexer to indexer, even if there _is_ a lot of overlap.

tl;dr: I'm cool with this change.

**Review by Jannis — APPROVED** [review]
